### PR TITLE
Update Property Data, Introduce `RevisedPropertyGenerator`, and Adjust Player Starting Money

### LINKED ISSUE #49: [49] General correction on the game's numerical system
This is an upgrade that increases the top level of properties from 3 to 5, and initial money from 1000 to 1500. The new numerical system better imitates the classic Monoply board game, which should ensure that the game will eventually end with a single winner not going bankrupt.

### PR DESCRIPTION
**Pull request by duduGrassUoft**
This pull request implements several changes to enhance gameplay and improve data consistency. It modifies the property data, introduces the `RevisedPropertyGenerator` class to adapt to the updated data format, and adjusts the initial money of players for improved logic flow.

### Changes Made:

**Property Data Update:**
- Modified the property data to increase the maximum level of properties from 3 to 5, allowing for deeper gameplay progression.

**`RevisedPropertyGenerator` Class:**
- Introduced the `RevisedPropertyGenerator` class to generate and initialize properties based on the updated data format.
- Defined methods to create properties with new attributes and levels.

 **Player Starting Money Adjustment:**
- Adjusted the initial money of players from 1500 to 100, aligning with the revised property values and avoiding potential logic bugs.

### CODE REVIEW

**Review by hecateli — APPROVED** [review]

**Review by AlwynYin — APPROVED** [review]

File: src/main/java/useCases/generator/RevisedPropertyGenerator.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            String line;
+            ArrayList<Integer> priceList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
+            ArrayList<Integer> taxList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
+
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(DELIMITER);
+
+                // generate the priceList, length 6 with last element being 0.
+                for (int i = 0; i < 5; i++) {
+                    priceList.set(i, Integer.parseInt(values[i + 2]));
```
Comment (NeverSoGoodStart) [inline src/main/java/useCases/generator/RevisedPropertyGenerator.java:36]: I think this is great, the original one only supports up to 3 different levels, which is probably not how Monopoly works

File: src/main/java/useCases/generator/RevisedPropertyGenerator.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * It is a generator generating the property blocks
+ */
+public class RevisedPropertyGenerator {
+
+    static final String DELIMITER = ",";
```
Comment (NeverSoGoodStart) [inline src/main/java/useCases/generator/RevisedPropertyGenerator.java:16]: Do we want to use the RegEx delimiter as used in `InitController`?
This one
",(?=([^\"]*\"[^\"]*\")*[^\"]*$)"

File: data/develop/revised_properties.csv [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+116,Knox College,220,150,150,150,150,18,90,250,700,875
+117,Canadiana Gallery,240,150,150,150,150,20,100,300,750,925
+119,University College,260,150,150,150,150,22,110,330,800,975
+120,Hart House,260,150,150,150,150,22,110,330,800,975
+121,Wycliffe College,280,150,150,150,150,24,120,360,850,1025
+122,Trinity College,300,200,200,200,200,26,130,390,900,1100
+123,Massey College,300,200,200,200,200,26,130,390,900,1100
+124,Rotman,320,200,200,200,200,28,150,450,1000,1200
+126,Innis College,350,200,200,200,200,35,175,500,1100,1300
+127,Roberts Library,400,200,200,200,200,50,200,600,1400,1700
```
Comment (NeverSoGoodStart) [inline data/develop/revised_properties.csv:22]: I think it's actually spelt as: 
"Rob**a**rts"

**Review by NeverSoGoodStart — APPROVED** [review]
Great change!
Just some small issues that we can probably fix in other commits, or if you have time, you can fix it straight up

**Pull request merged**